### PR TITLE
Wish list page 07

### LINKED ISSUE #13: feat: Wishlist Page

### PR DESCRIPTION
**Pull request by sadathshariff**
Wishlist Page Implemented
Add To Wishlist
Remove From wishlist
Move to wishlist
move from Wishlist To Cart
Update the Qty if  already Present

### CODE REVIEW

File: src/components/ProductDetails/ProductDetail.jsx [hunk 1]
```diff
@@ -21,20 +23,29 @@ export const ProductDetail = () => {
...
+  // const addToWishlist = async (product) => {
+  //   try {
+  //     const newWishlist = await postToWishlist(product, token);
+  //     if (newWishlist) {
+  //       wishlistDispatch({ type: "ADD_TO_WISHLIST", payload: newWishlist });
+  //     }
+  //   } catch (error) {
+  //     console.error(error);
+  //   }
+  // };
```
Comment (nishasen) [inline src/components/ProductDetails/ProductDetail.jsx]: Remove the commented code

File: src/utils/cart.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  }
+};
+
+export const removeFromCart = async (id, cartDispatch, deleteFromCart) => {
+  try {
+    const deletedItem = await deleteFromCart(id, token);
+    if (deletedItem) {
+      cartDispatch({ type: "REMOVE_FROM_CART", payload: deletedItem });
+    }
+  } catch (err) {
```
Comment (nishasen) [inline src/utils/cart.js:19]: try calling dispatch after api call only, you dont have to do try catch two time then

File: src/components/Card/WishlistCard/WishlistCard.jsx [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+            <FaRegHeart
+              onClick={() =>
+                addToWishlist(product, wishlistDispatch, postToWishlist)
+              }
+            />
+          )}
+        </div>
+        <Link to="/productDetails">
+          <div className="card-header">
+            <h3 className="headline-3">{product.name}</h3>
```
Comment (krishanu2911) [inline src/components/Card/WishlistCard/WishlistCard.jsx:48]: Just a small suggestion that you can destructure these Values above

**Review by nishasen — APPROVED** [review]
I liked your api call approaches, Good work. Have suggested small changes else everything looks good.

**Review by krishanu2911 — APPROVED** [review]
Everything is good to go. Well written code👍🏻
Just added a small suggestion

File: src/utils/cart.js (hunk 2 above)
Comment (sadathshariff) [reply src/utils/cart.js:19]: Thanks for that

File: src/components/Card/WishlistCard/WishlistCard.jsx (hunk 3 above)
Comment (sadathshariff) [reply src/components/Card/WishlistCard/WishlistCard.jsx:48]: yeah but i have to pass entire product onClick so did this way

**Pull request merged**